wasm/engine.go

```diff
@@ -9,4 +9,8 @@ type Engine interface {
 	Call(f *FunctionInstance, args ...uint64) (returns []uint64, err error)
 	// Compile compiles down the function instance.
 	Compile(f *FunctionInstance) error
+	// PreCompile prepares the compilation for given function instances.
```

Comment:
maybe add why? you have what done

Reply:
also, this seems very dangerous to invoke twice...

Reply:
https://github.com/tetratelabs/wazero/pull/60/commits/d160135e0066a9b470dad6d7e22b3f579bb0300b done